src/Tests/Reproduce/GithubIssue2323.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Linq;
+using FluentAssertions;
+using Nest;
+using Tests.Framework;
+using Tests.Framework.Integration;
+using Tests.Framework.MockData;
+using Xunit;
+
+namespace Tests.Reproduce
+{
+	public class GithubIssue2323 : IClusterFixture<ReadOnlyCluster>
```

Comment:
Nitpick: `_nested` serialization is covered by this integration test but it'd be good to add a unit test for this too